tests/unit/test_charm.py

```diff
@@ -677,7 +678,8 @@ def test_exec_error_on_client_changed_event_emitted(
         relation_id=relation_id, client_id="client_id", **CLIENT_CONFIG
     )
 
-    assert caplog.record_tuples[0][2] == f"Exited with code: {err.exit_code}. Stderr: {err.stderr}"
+    error_messages = [record[2] for record in caplog.record_tuples]
```

Comment:
for next time, if this is not necessary for the purpose/scope of the PR better to leave it out and make another one

Reply:
it was necessary, because adding the tracing code was breaking this unit test.